src/java_bytecode/java_object_factory.cpp

```diff
@@ -411,6 +430,247 @@ void java_object_factoryt::gen_pointer_target_init(
 
 /*******************************************************************\
 
+Function: java_object_factoryt::gen_nondet_pointer_init
+
+  Inputs: `assignemnts` - the code block we are building with initilisation code
+          `expr`: lvalue expression to initialise
+          `create_dynamic_objects`: if true, use malloc to allocate
+            objects; otherwise generate fresh static symbols.
+          `pointer_type` - The type of the pointer we are initalising
+          `update_in_place`:
+            NO_UPDATE_IN_PLACE: initialise `expr` from scratch
+            MUST_UPDATE_IN_PLACE: reinitialise an existing object
+            MAY_UPDATE_IN_PLACE: generate a runtime nondet branch
+              between the NO_ and MUST_ cases.
+
+ Outputs:
+
+ Purpose: Initialises a primitive or object tree rooted at `expr`,
+          of type pointer. It allocates child objects as necessary
+          and nondet-initialising their members, or if MUST_UPDATE_IN_PLACE is
+          set, re-initialising already-allocated objects.
+
+\*******************************************************************/
+
+void java_object_factoryt::gen_nondet_pointer_init(
+  code_blockt &assignments,
+  const exprt &expr,
+  const irep_idt &class_identifier,
+  bool create_dynamic_objects,
+  const pointer_typet &pointer_type,
+  const update_in_placet &update_in_place)
+{
+  const typet &subtype=ns.follow(pointer_type.subtype());
+  if(subtype.id()==ID_struct)
+  {
+    const struct_typet &struct_type=to_struct_type(subtype);
+    const irep_idt struct_tag=struct_type.get_tag();
+    // If this is a recursive type of some kind, set null.
+    if(!recursion_set.insert(struct_tag).second)
+    {
+      if(update_in_place==NO_UPDATE_IN_PLACE)
+      {
+        assignments.copy_to_operands(
+          get_null_assignment(expr, pointer_type));
+      }
+      // Otherwise leave it as it is.
```

Comment:
Comment is from [original code](https://github.com/diffblue/cbmc/pull/990/files/9ca22853f62d418adeeab046bf8e1f5fcba77c02#diff-233653d9ec20537d004eef631225bfffL478). I don't _think_ there is supposed to be an else case. I read "leave as it is" to mean there is nothing to be done, can you think of a wording that doesn't imply to you an else clause?